src/Microsoft.AspNetCore.Mvc.Core/UrlHelperExtensions.cs

```diff
@@ -440,5 +442,23 @@ public static string Page(this IUrlHelper urlHelper, string pageName)
                 host: host,
                 fragment: fragment);
         }
+
+        private static object CalculatePageName(ActionContext actionContext, string pageName)
```

Comment:
We sort of already had the infrastructure to test the Page method directly. Seemed convenient to test it in entirety.